### PR TITLE
feat: Add columns to table

### PR DESCRIPTION
**Pull request by VanTekken**
Added functionality necessary to add a column to an existing table. Additional development could be done in this area, such as listing existing columns for databases and checking whether a column already exists before executing. Furthermore, additional column data types could easily be added in future.

### CODE REVIEW

**Comment by VanTekken** [comment]
Reopened pull request after removing test database file. Tried adding "feat:" to title and commit to satisfy Semantic PR test, though this has still failed. 🥷

**Comment by VanTekken** [comment]
#16

File: routes/handlers.go [hunk 1]
```diff
@@ -81,3 +81,21 @@ func DropTable(c *fiber.Ctx) error {
...
 	return c.RedirectBack("/")
 }
+
+/*
+	AddColumn takes the table name from the post request
+
+	and calls the AddColumn of the db package.
+*/
+func AddColumn(c *fiber.Ctx) error {
+	table := new(TableStruct)
```
Comment (Shivansh-yadav13) [inline routes/handlers.go:91]:
```suggestion
	table := TableStruct{}
```

File: routes/handlers.go [hunk 2]
```diff
@@ -81,3 +81,21 @@ func DropTable(c *fiber.Ctx) error {
...
+	column := new(ColumnStruct)
```
Comment (Shivansh-yadav13) [inline routes/handlers.go:92]:
```suggestion
	column := ColumnStruct{}
```

File: routes/handlers.go [hunk 3]
```diff
@@ -81,3 +81,21 @@ func DropTable(c *fiber.Ctx) error {
...
+	if err := c.BodyParser(table); err != nil {
```
Comment (Shivansh-yadav13) [inline routes/handlers.go:93]:
```suggestion
	if err := c.BodyParser(&table); err != nil {
```

Comment (Shivansh-yadav13) [reply routes/handlers.go:93]: also we need to parse for column struct
```go
if err := c.BodyParser(&column); err != nil {
		return err
}
```

File: routes/handlers.go [hunk 4]
```diff
@@ -81,3 +81,21 @@ func DropTable(c *fiber.Ctx) error {
...
+		return err
+	}
+	if err := db.AddColumn(table.Name, column.Name, column.Type); err != nil {
+		log.Fatalln(err)
+	}
+
+	return c.RedirectBack("/")
```
Comment (Shivansh-yadav13) [inline routes/handlers.go:100]:
```suggestion
	return c.RedirectBack(fmt.Sprintf("/%s", table.Name))
```

File: templates/table_view.html [hunk 5]
```diff
@@ -9,5 +9,16 @@
 </head>
 <body>
   {{.table_name}}
+  <form action="/add-column" method="post">
+    <label for="table_name">&#43; Add Column</label>
+    <input type="hidden" id="table_name" name="table_name" placeholder="table-name" value="{{.table_name}}"/>
+    <input type="text" id="column_name" name="column_name" placeholder="column-name"/>
+    <label for="data_type"></label>
+    <select name="column_type">
+      <option value="CHARACTER(20)">CHARACTER(20)</option>
```
Comment (Shivansh-yadav13) [inline templates/table_view.html:18]: can you add more datatypes like int etc?

**Review by Shivansh-yadav13 — CHANGES_REQUESTED** [review]

**Comment by Shivansh-yadav13** [comment]
> Tried adding "feat:" to title and commit to satisfy Semantic PR test, though this has still failed

maybe bcuz one commit is still not semantic you can squash the commits to fix it.

File: templates/table_view.html (hunk 5 above)
Comment (VanTekken) [reply templates/table_view.html:18]: Sure! Added integer, float and boolean data types 🥷

**Comment by VanTekken** [comment]
Commits have been squashed to be more concise and satisfy Semantic PR checks.

File: routes/handlers.go [hunk 6]
```diff
@@ -81,3 +81,21 @@ func DropTable(c *fiber.Ctx) error {
...
+	table := TableStruct{}
+	column := ColumnStruct{}
+	if err := c.BodyParser(&table); err != nil {
+		return err
+	}
+	if err := db.AddColumn(table.Name, column.Name, column.Type); err != nil {
+		log.Fatalln(err)
+	}
+
+	return c.RedirectBack(fmt.Sprintf("/%s", table.Name))
```
Comment (Shivansh-yadav13) [inline routes/handlers.go:100]: @VanTekken you forgot to import `"fmt"` in the import statement

File: routes/handlers.go [hunk 7]
```diff
@@ -81,3 +81,21 @@ func DropTable(c *fiber.Ctx) error {
...
+	AddColumn takes the table name from the post request
+
+	and calls the AddColumn of the db package.
+*/
+func AddColumn(c *fiber.Ctx) error {
+	table := TableStruct{}
+	column := ColumnStruct{}
+	if err := c.BodyParser(&table); err != nil {
+		return err
+	}
```
Comment (Shivansh-yadav13) [inline routes/handlers.go:95]:
```suggestion
	}
	if err := c.BodyParser(&column); err != nil {
		return err
	}
```

Everything is working :+1:  you just need to parse the column name and type from the request.

**Review by Shivansh-yadav13 — APPROVED** [review]
Just make the above requested changes. LGTM :+1:

**Comment by VanTekken** [comment]
Changes made. Let me know if anything else is needed 🥷

**Comment by Shivansh-yadav13** [comment]
@VanTekken IG you added other changes also in the commit just revert it to old commit `30ac711` then add the `"fmt"` without the squash and merge I will merge the changes.

**Comment by VanTekken** [comment]
Sorry, I added and then removed fmt.Println statements to debug the output of parsed values. Please check if the revert and addition of fmt package is ok!

**Pull request merged**